haystack/components/validators/json_schema.py

```diff
@@ -0,0 +1,172 @@
+import json
+from typing import List, Any, Dict, Optional
+
+from haystack import component
+from haystack.dataclasses import ChatMessage
+from haystack.lazy_imports import LazyImport
+
+with LazyImport(message="Run 'pip install jsonschema'") as jsonschema_import:
+    from jsonschema import validate, ValidationError
+
+
+@component
+class JsonSchemaValidator:
```

Comment:
This name sounds generic, while the component probably is not.
- Is it specific for Chat Messages?
- Is it specific for Function Calling?

Let's try to understand the purpose and choose a more expressive name.

Reply:
What do you suggest? I think we should keep the ChatMessage signature but enable JSON validation for all kinds of messages carrying JSON payload - i.e., only function calling. As LLM json capabilities increase users will likely be creating lots of json needing validation and recovery loops

Reply:
`ChatMessageJsonValidator`?

Reply:
Totally open to suggestions here. Let's have an internal discussion.